feat(frontend): add referral modal

boriskubrik opened the pull request:
- Adds a referral modal. 
- Places links in the Menu and in the footer.
- The URL does not contain `feeBps`, sticking to default. UI shows 1% because 50% goes to the protocol.
- Uses context to keep track of the state as we have the modal in multiple places.
- Triggers the share action on mobile, copies to clipboard on desktop.
- Shows toast after successful copy on desktop.
- Shows connect wallet if the user is not connected.

### Testing
- [x] Sharing works on mobile using MetaMask
- [x] Copy works on desktop
- [x] Referral address and fee set in localstorage after using the link
- [x] Connect wallet state works

### Screenshot 
![image](https://github.com/sifiorg/sifi/assets/128667801/02b67bb4-089f-4191-bbbb-4ea681a21ac6)

tommy0x commented on packages/frontend/src/modals/ReferralModal/ReferralModal.tsx:
```diff
@@ -0,0 +1,69 @@
+  const generateReferralLink = () => {
+    if (!address) return null;
+
+    const feeBps = 20;
```
This should be 200 if I'm correct. 1 BPS is 0.01%

immutablegoblin reviewed (commented):
Looks good, just needs bps adjustment as noted by Tommy

boriskubrik commented:
Need to hold on until the URL structure is finalized. #479

xykota commented:
Let's make this ref link not include the fee bps. Then we'll have the default fee when there is a partner at 40 bps and without a partner 20 bps.

Then in the next iteration we will let the user customize the fee bps and reflect in the text they are sharing it down the middle with the protocol.

ref #479

boriskubrik commented:
Rebased, updated the URL, tested and it works. Ready for review.

![image](https://github.com/sifiorg/sifi/assets/128667801/c059c542-c707-4001-9957-c160b2b21030)

immutablegoblin commented:
It would be nice not to have the linebreak when the modal is at full width if possible?

Working well though.

![image](https://github.com/sifiorg/sifi/assets/112887817/3180211e-aa82-4e7c-b3f9-3c985b577419)

boriskubrik commented:
@immutablegoblin switched to `break-all`.

![image](https://github.com/sifiorg/sifi/assets/128667801/3a16d825-7797-416a-bab3-0ecfb2e9c0c5)
